Library/Tracking/KSRAnalyticsTests.swift

```diff
@@ -899,8 +899,29 @@ final class KSRAnalyticsTests: TestCase {
 
     ksrAnalytics.trackPledgeCTAButtonClicked(stateType: .pledge, project: project)
 
-    XCTAssertEqual(["Project Page Pledge Button Clicked"], dataLakeClient.events)
-    XCTAssertEqual(["Project Page Pledge Button Clicked"], segmentClient.events)
+    XCTAssertEqual(["CTA Clicked"], dataLakeClient.events)
+    XCTAssertEqual(["CTA Clicked"], segmentClient.events)
+  }
+
+  func testTrackPledgeCTAButtonClicked_Context() {
+    let dataLakeClient = MockTrackingClient()
+    let segmentClient = MockTrackingClient()
+    let project = Project.template
+    let loggedInUser = User.template |> \.id .~ 42
+
+    let ksrAnalytics = KSRAnalytics(
+      dataLakeClient: dataLakeClient,
+      loggedInUser: loggedInUser,
+      segmentClient: segmentClient
+    )
+
+    ksrAnalytics.trackPledgeCTAButtonClicked(stateType: .pledge, project: project)
+
+    let dataLakeClientProps = dataLakeClient.properties.last
```

Comment:
I think you can move line 920-924 into the method above this. The tests are doing the same thing with the exception of the properties check so it's ok to consolidate.